buildSrc/src/main/kotlin/SystemTestUtils.kt

```diff
@@ -0,0 +1,31 @@
+import org.gradle.api.tasks.testing.Test
+
+/**
+ * Utilities for system tests.
+ */
+object SystemTestUtils {
+
+	private var nextFreePort = 6000
+	private val lock = Object()
+
+	/**
+	 * Used ports should be unique for each system test to avoid collisions during parallel execution.
+	 * Thus, we assign a new port each time this is called.
+	 * Since tests may run in different workers that don't share the #nextFreePort variable, we use the worker number
+	 * to avoid collisions (given that we don't have more than 100 system tests per worker).
+	 */
+	fun pickFreePort(): Int {
+		synchronized(lock) {
+			val pickedPort = nextFreePort
+			nextFreePort += 1
+			return pickedPort
+		}
+	}
+
+}
+
+/** The port that the fake Teamscale should use during the system test. Guaranteed conflict-free. */
+val Test.teamscalePort: Int by lazy { SystemTestUtils.pickFreePort() }
```

Comment:
Very nice idea!
Unfortunately this won't work reliably, because Gradle already uses classloader isolation for subprojects that do not apply the exact same set of plugins, which is already the case for some system test modules. This will also be used more in the future with configuration caching and project isolation being enabled. The only reliable solution for this problem is to make `SystemTestUtils` a [shared build service](https://docs.gradle.org/current/userguide/build_services.html) (see also the related discussion on [Github](https://github.com/gradle/gradle/issues/14697))

Reply:
I did my best but I'm not 100% sure I implemented this build service right. The docs are all about using the service in custom tasks but we're modifying the existing Test task, so that doesn't seem to apply. At least in my local setup, it picks all distinct ports, but I don't think it ran anything in parallel anyways, so better review this extra carefully ;-)